app/models/host/managed_extensions.rb

```diff
@@ -61,4 +61,9 @@ def delete_discovery_attribute_set
     DiscoveryAttributeSet.destroy_all(:host_id => self.id) if type_changed?
   end
 
+  def update_notifications
+    return if new_record?
+    return unless type_changed?
+    ForemanDiscovery::UINotifications::DestroyHost.deliver!(self)
```

Comment:
this in theory will break running all core tests with foreman's as this extends Host::Managed yet the seeding happens only in discovery, I'm uncertain if this is an issue or not, but if it is we might need to change the way blueprint seeds is loaded in core. (TBH I don't think it should block this pr).